### ISSUE TITLE
Float16 does not work on NVIDIA Titan V

### ISSUE THREAD

**Issue by neil-119** [keras-team/keras#10262]
Float16 gets stuck at ~10% accuracy on the NVIDIA Titan V. Please see gist and output below:
https://gist.github.com/neil-119/0493832eb5ecb0387943d69e4691b859

Works on PyTorch and MXNet.

> python sample.py
> /home/nrao/anaconda3/envs/tensorflow/lib/python3.4/importlib/_bootstrap.py:321: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
>   return f(*args, **kwds)
> Using TensorFlow backend.
> x_train shape: (60000, 28, 28, 1)
> 60000 train samples
> 10000 test samples
> Train on 60000 samples, validate on 10000 samples
> Epoch 1/12
> 2018-05-22 09:55:04.288476: I tensorflow/core/platform/cpu_feature_guard.cc:140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
> 2018-05-22 09:55:04.452269: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:898] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
> 2018-05-22 09:55:04.452738: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1356] Found device 0 with properties:
> name: TITAN V major: 7 minor: 0 memoryClockRate(GHz): 1.455
> pciBusID: 0000:01:00.0
> totalMemory: 11.78GiB freeMemory: 10.76GiB
> 2018-05-22 09:55:04.452755: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1435] Adding visible gpu devices: 0
> 2018-05-22 09:55:04.636799: I tensorflow/core/common_runtime/gpu/gpu_device.cc:923] Device interconnect StreamExecutor with strength 1 edge matrix:
> 2018-05-22 09:55:04.636842: I tensorflow/core/common_runtime/gpu/gpu_device.cc:929]      0
> 2018-05-22 09:55:04.636847: I tensorflow/core/common_runtime/gpu/gpu_device.cc:942] 0:   N
> 2018-05-22 09:55:04.637045: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1053] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 10397 MB memory) ->physical GPU (device: 0, name: TITAN V, pci bus id: 0000:01:00.0, compute capability: 7.0)
> 60000/60000 [==============================] - 4s 68us/step - loss: nan - acc: 0.1031 - val_loss: nan - val_acc: 0.0980
> Epoch 2/12
> 60000/60000 [==============================] - 3s 51us/step - loss: nan - acc: 0.0987 - val_loss: nan - val_acc: 0.0980
> Epoch 3/12
> 60000/60000 [==============================] - 3s 48us/step - loss: nan - acc: 0.0987 - val_loss: nan - val_acc: 0.0980

**Comment by neil-119** [comment]
I just tested the same code on Windows, and same issue.

**Comment by ialhashim** [comment]
Was your code working on other cards? I started having issues when we upgraded to Titan V [(issue)](https://github.com/tensorflow/tensorflow/issues/19290)

I tried upgrading everything (except for cuda.. could be the issue??).

**Comment by neil-119** [comment]
I haven't tried this code, but float 16 was working fine on my GTX 1080 (albeit much slower). I was hoping to take advantage of fp16 performance on Volta GPUs, but it seems I'll have to rely on PyTorch or MXNet for the time being.

**Comment by raghavgurbaxani** [comment]
@neil-119  did you manage to solve the issue ? I am facing the same problem ..

**Comment by ChrisBellew** [comment]
I am also experiencing this issue on AWS p3.8xlarge machines

**Comment by neil-119** [comment]
@raghavgurbaxani  No, and I didn't have time to delve into the backend code for Keras, so I switched to another framework.
